### PR TITLE
feat(cb2-8099): save state for accordion sections

### PR DESCRIPTION
**Pull request by shivangidas**
## Tech record summary sections remain open after page changes

Save the state of whether a section is expanded to not, and keep it open after navigation from tyre search, vin/vrm change, batch vehicles, etc.

[CB2-8099](https://dvsa.atlassian.net/browse/CB2-8099)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <section class="govuk-grid-row">
-  <app-accordion-control [class]="'padding'">
+  <app-accordion-control [class]="'padding'" [sectionState]="sectionTemplateState">
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.html:2]:
```suggestion
  <app-accordion-control [class]="'padding'" [sectionState]="sectionTemplateState$ | async">
```

File: src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
     <div class="govuk-grid-column-one-half">
       <ng-container *ngFor="let section of sectionTemplates; let i = index">
-        <app-accordion *ngIf="i < middleIndex" [id]="section.name" [title]="section.label">
+        <app-accordion *ngIf="i < middleIndex" [id]="section.name" [title]="section.label" [isExpanded]="isSectionExpanded(section.name)">
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.html:5]:
```suggestion
        <app-accordion *ngIf="i < middleIndex" [id]="section.name" [title]="section.label" [isExpanded]="isSectionExpanded$(section.name) | async">
```

**Review by gjulien-bjss — COMMENTED** [review]
All of the following should avoid managing subscriptions in the component

File: src/app/services/technical-record/technical-record.service.ts [hunk 3]
```diff
@@ -180,4 +182,11 @@ export class TechnicalRecordService {
   get viewableRecordStatus$(): Observable<StatusCodes | undefined> {
     return this.viewableTechRecord$.pipe(map(techRecord => techRecord?.statusCode));
   }
+
+  get sectionStates$(): Observable<(string | number)[] | undefined> {
+    return this.store.pipe(select(selectSectionState));
+  }
+  clearSectionTemplateStates() {
+    this.store.dispatch(clearAllSectionStates());
+  }
```
Comment (gjulien-bjss) [inline src/app/services/technical-record/technical-record.service.ts:191]:
```suggestion
  get sectionStates$(): Observable<(string | number)[] | undefined> {
    return this.store.pipe(select(selectSectionState));
  }
  
  clearSectionTemplateStates() {
    this.store.dispatch(clearAllSectionStates());
  }
```
nitpick

File: src/app/shared/components/accordion-control/accordion-control.component.ts [hunk 4]
```diff
@@ -35,9 +36,15 @@ export class AccordionControlComponent {
     this.cdr.markForCheck();
   }
 
+  private expandAccordians(): void {
```
Comment (gjulien-bjss) [inline src/app/shared/components/accordion-control/accordion-control.component.ts:39]:
```suggestion
  private expandAccordions(): void {
```

**Review by kcarr-bjss — APPROVED** [review]

**Review by owen-corrigan-bjss — APPROVED** [review]

**Pull request merged**